packages/compass-e2e-tests/helpers/commands/shell-eval.js

```diff
@@ -4,24 +4,36 @@ const Selectors = require('../selectors');
 module.exports = function (app) {
   return async function (str, parse = false, timeout = 10000) {
     const { client } = app;
-    if (!(await client.isVisible(Selectors.ShellContent))) {
+    const shellContentElement = await client.$(Selectors.ShellContent);
+    if (!(await shellContentElement.isDisplayed())) {
       await client.clickVisible(Selectors.ShellExpandButton);
     }
     await client.clickVisible(Selectors.ShellInput);
     // Might be marked with a deprecation warning, but can be used
     // https://github.com/webdriverio/webdriverio/issues/2076
+
     await client.keys(parse === true ? `JSON.stringify(${str})` : str);
     await client.keys('\uE007');
-    await client.waitUntil(
-      async () => {
-        return !(await client.isVisible(Selectors.ShellLoader));
-      },
-      timeout,
-      `Expected shell evaluation to finish in ${timeout}ms`,
-      50
-    );
+
+    const shellLoaderBarElement = await client.$(Selectors.ShellLoader);
+    if (await shellLoaderBarElement.isDisplayed()) {
+      await client.waitUntil(
+        async () => {
+          return !(await shellLoaderBarElement.isDisplayed());
+        },
+        {
+          timeout,
+          timeoutMsg: `Expected shell evaluation to finish in ${timeout}ms`,
+          interval: 50,
+        }
+      );
```

Comment:
Can it be replaced with this?

```suggestion
      await shellLoaderBarElement.waitForDisplayed({ 
        timeout,
        timeoutMsg: `Expected shell evaluation to finish in ${timeout}ms`,
        reverse: true,
        interval: 50
      })
```